tests/e2e/tour/tour.spec.js

```diff
@@ -9,10 +11,107 @@ const expectWithinMargin = async (actual, expected, margin) => {
+  test.beforeEach(async ({ page, context }) => {
+    await setupTests({
+      page,
+      context,
+      dontShowTour: false
+    })
+
+    await page.route(/collections.json/, async (route) => {
+      await route.fulfill({
+        json: singleCollection.body,
+        headers: singleCollection.headers
+      })
+    })
+
+    await page.goto('/search')
+  })
+
+  test.describe('When clicking the "Skip for now" button', () => {
+    test.beforeEach(async ({ page }) => {
+      // Expect the first step to show the "Take the tour" button
+      await expect(page.locator('.search-tour__welcome')).toContainText('Welcome to Earthdata Search!')
+
+      // Click the "Skip for now" button to close the tour
+      await page.click('button:has-text("Skip for now")')
+    })
+
+    test('should close the tour', async ({ page }) => {
+      // Ensure the tour is closed by checking that the tour container is no longer visible
+      await expect(page.locator('.search-tour__container')).toBeHidden()
+    })
+
+    test.describe('When refreshing the page', () => {
+      test('should show the tour again', async ({ page }) => {
+        // Refresh the page
+        await page.reload()
+
+        // Expect the tour to run on page load
+        await expect(page.locator('.search-tour__welcome')).toContainText('Welcome to Earthdata Search!')
+      })
+    })
+  })
+
+  test.describe('When checking the "Don\'t show again" checkbox', () => {
+    test.beforeEach(async ({ page }) => {
+      // Verify the tour is open
+      await expect(page.locator('.search-tour__welcome')).toContainText('Welcome to Earthdata Search!')
+
+      // Verify the checkbox is unchecked
+      const checkbox = page.getByRole('checkbox', { name: 'Don\'t show the tour next time I visit Earthdata Search' })
+      await expect(checkbox).not.toBeChecked()
+
+      // Check the checkbox and verify it is checked
+      await checkbox.click()
+      await expect(checkbox).toBeChecked()
+    })
+
+    test('should not see the tour when the page reloads if the checkbox is checked', async ({ page }) => {
+      await page.reload()
+
+      await page.locator('.sidebar-section__header-primary .sidebar-section__title', { hasText: 'Filter Collections' }).waitFor()
+
+      await expect(page.getByRole('alertdialog', { value: /Welcome to Earthdata Search/ })).toBeHidden()
+    })
+  })
+})
+
+test.describe('When loading the page with dontShowTour preference set to true', () => {
+  test.beforeEach(async ({ page, context }) => {
+    await setupTests({
+      page,
+      context,
+      dontShowTour: true
+    })
+
+    await page.route(/collections.json/, async (route) => {
+      await route.fulfill({
+        json: singleCollection.body,
+        headers: singleCollection.headers
+      })
+    })
+
+    await page.goto('/search')
+  })
+
+  test('should not see the tour when the page loads', async ({ page }) => {
+    await page.reload()
```

Comment:
This reload isn't necessary, the previous action taken in the beforeEach is page.goto()